### PR TITLE
[Filebeat] Add cloudwatch fileset in aws module

### LINKED ISSUE #13716: [Filebeat] Add support for AWS CloudWatch logs
The goal of this issue is to create a filebeat fileset to support AWS CloudWatch logs. Depends on the CloudWatch logs type, there might be some additional work on the s3 input needs to be done first.

## Filebeat module

* [ ] Test log files exist for the grok patterns
* [ ] Generated output for at least 1 log file exists

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
## What does this PR do?

This PR is to add cloudwatch fileset into aws module to parse logs from CloudWatch AWS. There are different kinds of logs can be sent to CloudWatch from different services. This PR is only focusing on EC2 logs. 

## Why is it important?

Users can use Amazon CloudWatch Logs to monitor, store, and access log files from Amazon EC2 instances, AWS CloudTrail, Route 53, and other sources. This fileset enables users to export logs into s3 bucket and parse logs using this fileset.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works

## Related issues

- Closes elastic/beats#13716

### CODE REVIEW

File: x-pack/filebeat/module/aws/cloudwatch/ingest/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+description: "Pipeline for s3 server access logs"
+
+processors:
+  - grok:
+      field: message
+      patterns:
+        - >-
+          %{TIMESTAMP_ISO8601:_tmp.timestamp} %{MONTH:_tmp.month} %{MONTHDAY:_tmp.day} %{TIME:_tmp.time} %{IPORHOST:aws.cloudwatch.ip} %{NOTSPACE:aws.cloudwatch.program_name}: %{GREEDYDATA:message}
```
Comment (mtojek) [inline x-pack/filebeat/module/aws/cloudwatch/ingest/pipeline.yml]: The field `aws.cloudwatch.ip` suggest it's only IP address, but according to the field definition it's also host. Is there any other field name that would cover both cases? Maybe `aws.cloudwatch.source` ?

File: filebeat/docs/fields.asciidoc [hunk 2]
```diff
@@ -1309,6 +1309,33 @@ type: keyword
...
++
+--
+The internet address of the requester.
+
+
+type: ip
+
+--
+
+*`aws.cloudwatch.program_name`*::
```
Comment (mtojek) [inline filebeat/docs/fields.asciidoc]: Is it the log group? If so, maybe keep the original naming?

File: x-pack/filebeat/module/aws/cloudwatch/ingest/pipeline.yml (hunk 1 above)
Comment (kaiyan-sheng) [reply x-pack/filebeat/module/aws/cloudwatch/ingest/pipeline.yml]: Thanks! `aws.cloudwatch.source` sounds better!

File: filebeat/docs/fields.asciidoc (hunk 2 above)
Comment (kaiyan-sheng) [reply filebeat/docs/fields.asciidoc]: It's not log group 😬 It is `systemd` part in `2020-02-20T07:01:01.000Z Feb 20 07:01:01 ip-172-31-81-156 systemd: Stopping User Slice of root`.

Comment (mtojek) [reply filebeat/docs/fields.asciidoc]: I see... hmm, that suggests "service_name", "application_name", but I'm fine with program too :)

File: x-pack/filebeat/module/aws/cloudwatch/ingest/pipeline.yml (hunk 1 above)
Comment (kaiyan-sheng) [reply x-pack/filebeat/module/aws/cloudwatch/ingest/pipeline.yml]: For example in `2020-02-20T07:01:01.000Z Feb 20 07:01:01 ip-172-31-81-156 systemd: Stopping User Slice of root.`, the ip part is `ip-172-31-81-156`. It's not in `172.31.81.156` format and that's why I chose IPORHOST to parse the field. So it covers a-z and -.

Comment (mtojek) [reply x-pack/filebeat/module/aws/cloudwatch/ingest/pipeline.yml]: That's alright ;) I was just referring to the field name, field def is ok.

Comment (kaiyan-sheng) [reply x-pack/filebeat/module/aws/cloudwatch/ingest/pipeline.yml]: Thanks for the review!!

**Comment by exekias** [comment]
I understand cloudwatch is more like an input, right @kaiyan-sheng? Where the actual logs (coming from log groups) can be of many kinds.

Would it make sense to create specific filesets for these different cloudwatch logs sources? For instance, the examples you are putting here probably map to the system module, that can be reused if configured with the right input.

Then we have a plethora of other services that are AWS specific and we don't yet support as filesets.

**Comment by kaiyan-sheng** [comment]
> I understand cloudwatch is more like an input, right @kaiyan-sheng? Where the actual logs (coming from log groups) can be of many kinds.

Yes, cloudwatch is more like an input. Different services can send logs into CW.

> Would it make sense to create specific filesets for these different cloudwatch logs sources? For instance, the examples you are putting here probably map to the system module, that can be reused if configured with the right input.

That would be better for sure! The examples are logs sent to CloudWatch from EC2 instance. Should we rename this fileset something like `cloudwatch_ec2`?

> Then we have a plethora of other services that are AWS specific and we don't yet support as filesets.
Great! That's why I added `%{TIMESTAMP_ISO8601:_tmp.timestamp} %{GREEDYDATA:message}` in grok pattern.

File: filebeat/docs/fields.asciidoc (hunk 2 above)
Comment (kaiyan-sheng) [reply filebeat/docs/fields.asciidoc]: I think I got this program_name from some AWS CLI. If I find it, I will link it here 😬

**Comment by exekias** [comment]
> That would be better for sure! The examples are logs sent to CloudWatch from EC2 instance. Should we rename this fileset something like `cloudwatch_ec2`?

That sounds great! I wonder if we should keep the “cloudwatch” prefix or do just `ec2`. At the end, I foresee these filesets being shared between Filebeat & Functionbeat at some point (through the integrations project). If you call it `ec2` now you can just make it clear in the docs & configs that this one comes from cloudwatch

**Comment by kaiyan-sheng** [comment]
> That sounds great! I wonder if we should keep the “cloudwatch” prefix or do just `ec2`. At the end, I foresee these filesets being shared between Filebeat & Functionbeat at some point (through the integrations project). If you call it `ec2` now you can just make it clear in the docs & configs that this one comes from cloudwatch

Yeah if we ever decides to add cloudwatch as a separate Filebeat input, these filesets can be shared there too. I will change the fileset name to `ec2` in next commit. Thanks! Will you be ok with `cloudwatch` as the fileset name for the default `%{TIMESTAMP_ISO8601:_tmp.timestamp} %{GREEDYDATA:message}` format logs?

**Comment by exekias** [comment]
> Yeah if we ever decides to add cloudwatch as a separate Filebeat input, these filesets can be shared there too. I will change the fileset name to `ec2` in next commit. Thanks! Will you be ok with `cloudwatch` as the fileset name for the default `%{TIMESTAMP_ISO8601:_tmp.timestamp} %{GREEDYDATA:message}` format logs?

I think this is a good idea!

File: filebeat/docs/fields.asciidoc (hunk 2 above)
Comment (kaiyan-sheng) [reply filebeat/docs/fields.asciidoc]: I decide to change it to process.name and process.pid to match what system filebeat module has.

**Comment by kaiyan-sheng** [comment]
@exekias @mtojek This PR is ready for another review (when you get a chance). Thanks!

**Comment by exekias** [comment]
btw this needs a make update

**Pull request merged**